web/src/message_list_view.js

```diff
@@ -619,101 +649,119 @@ export class MessageListView {
         return undefined;
     }
 
-    build_message_groups(message_containers) {
-        const start_group = () => ({
-            message_containers: [],
-            message_group_id: _.uniqueId("message_group_"),
-        });
+    build_message_groups(messages) {
+        let group_message_containers = [];
 
-        let current_group = start_group();
         const new_message_groups = [];
+        let current_group;
+
         let prev;
 
         const add_message_container_to_group = (message_container) => {
-            current_group.message_containers.push(message_container);
+            group_message_containers.push(message_container);
         };
 
-        const finish_group = () => {
-            if (current_group.message_containers.length > 0) {
-                populate_group_from_message_container(
-                    current_group,
-                    current_group.message_containers[0],
-                );
+        const finish_group = (prev_message, message_for_next_group) => {
```

Comment:
maybe rename to `finish_group_and_start_new_group`? or separate into two functions